### PR TITLE
Regression suite enhancement

### LINKED ISSUE #394: Regression test suite cannot be run multiple times consecutively in local installs
Fails on model registration tests as the mar already exists in the model store. So, we can either
1. Have instructions to restart the torchserve in a new model_store folder if running collection from PostMan client.
2. Rename the mar file downloaded with a timestamp so it doesn't clash with currently existing file when model registration API is invoked

### PR DESCRIPTION
**Pull request by shivamshriwas**
## Description

- fixes issue for rerun test suite consecutively.
- dumps serve logs in  /tmp/ts.log for all collections run.
- cleans test suite logs for each run
-  Cleanup after each model test  
-  added response comparison for text based and json (image classification json output) for others and binary output , status code is verified.
- JSON output scores is compared with tolerance factor given in data file. As output scores of model  will have prediction may vary.
- Enhanced the regression tests added for huggingface examples

Fixes #394 #393 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

Run regression suite consecutively
- [x] Test A  
Regression suite runs consecutively

- [X] Test B
for model zoo inference test models are unregistered after inference test is completed of the model.
Verify in /tmp/test_exec.log

- UT/IT execution results

- Logs
Regression suite was run consecutively twice in a docker container and below are generated console logs and reports.
Reports names are renamed here, originally `test_exec.log` and `ts.log` are file names and they are overwritten if the suite is rerun. instance 
  - console logs : [regression_suite_enhacement_console_log.txt](https://github.com/pytorch/serve/files/4820130/regression_suite_enhacement_console_log.txt)

  - Reports:
 Test suite execution logs:
 [test_exec_2.log](https://github.com/pytorch/serve/files/4820126/test_exec_2.log)
 [test_exec_1.log](https://github.com/pytorch/serve/files/4820127/test_exec_1.log)

  - Test suite execution server logs:
 [ts_2.log](https://github.com/pytorch/serve/files/4820128/ts_2.log)
 [ts_1.log](https://github.com/pytorch/serve/files/4820129/ts_1.log)

Tested multiple runs of regression suite on the following instance as well:
1. GPU - EC2 gpu docker container[  gpu_device_gpu_docker.zip](https://github.com/pytorch/serve/files/4860665/gpu_device_gpu_docker.zip)
2. GPU - EC2  CPU docker container   [gpu_device_cpu_docker.zip](https://github.com/pytorch/serve/files/4860666/gpu_device_cpu_docker.zip)
3. CPU - EC2 CPU docker container  [cpu_device_cpu_docker.zip](https://github.com/pytorch/serve/files/4860667/cpu_device_cpu_docker.zip)

## Currrent know test case failures
Model zoo BERT scripted models inference test cases are failing on GPU instance type. : #479
         Once the new BERT mar files are updated on S3 these test cases will pass on GPU as well
VGG11 fails on GPU : #535
Management API failure: #511

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

Note: This is the rebased PR of #398

### CODE REVIEW

**Comment by dhanainme** [comment]
shivamshriwas@ can you please add some details about testing was performed for this. 
Specifically logs for regression test suite in a Docker container.

**Comment by shivamshriwas** [comment]
@dhanainme I have updated the logs in the PR description.

File: test/regression_tests.sh [hunk 1]
```diff
@@ -28,9 +28,6 @@ install_torchserve_from_source() {
   cd serve
   echo "Installing torchserve torch-model-archiver from source"
   ./scripts/install_from_src_ubuntu
-  echo "TS Branch : " $(git rev-parse --abbrev-ref HEAD) >> $3
-  echo "TS Branch Commit Id : " $(git rev-parse HEAD) >> $3
-  echo "Build date : " $(date) >> $3
   echo "Torchserve Succesfully installed"
```
Comment (dhanainme) [inline test/regression_tests.sh:31]: This info was recently added to track date / branch / commit against which the test ran. Please dont remove this or add this elsewhere.

File: test/postman/inference_data.json [hunk 2]
```diff
@@ -1,120 +1,360 @@
...
+            {
+                "Egyptian_cat":0.11440037935972214
+            },
+            {
+                "lynx":0.006006305105984211
+            },
+            {
+                "tiger":0.004394155461341143
+            }
+        ],
```
Comment (dhanainme) [inline test/postman/inference_data.json:143]: These numbers are too specific even for a test. 

Can you please add some info about testing done to confirm that the tolerance values are in the range that we expect.

Comment (shivamshriwas) [reply test/postman/inference_data.json:143]: Added testing logs of the regression suite tested on the various execution environments(updated PR description).
The set tolerance value works fine on following 
1.  CPU device local execution
2. GPU  device local execution
3. CPU device, cpu docker container
4. GPU device, cpu docker container
5. GPU device, gpu docker container

File: test/regression_tests.sh (hunk 1 above)
Comment (shivamshriwas) [reply test/regression_tests.sh]: updated as required

**Comment by shivamshriwas** [comment]
Model zoo BERT scripted models inference test cases are failing on GPU instance type. 
Which are dependent  on the issue : https://github.com/pytorch/serve/issues/479

**Comment by shivamshriwas** [comment]
New failures in test suite due to the following issues,

VGG11 fails on GPU : https://github.com/pytorch/serve/issues/535
Management API failure: https://github.com/pytorch/serve/issues/511

**Comment by dhanainme** [comment]
Merged as #549

**Pull request merged**